src/Test/test_attribute.jl

```diff
@@ -190,6 +190,72 @@ function setup_test(
     return
 end
 
+"""
+    test_attribute_AbsoluteGapTolerance(model::MOI.AbstractOptimizer, config::Config)
+
+Test that the [`MOI.AbsoluteGapTolerance`](@ref) attribute is implemented for
+`model`.
+"""
+function test_attribute_AbsoluteGapTolerance(
+    model::MOI.AbstractOptimizer,
+    ::Config,
+)
+    @requires MOI.supports(model, MOI.AbsoluteGapTolerance())
+    # Get the current value to restore it at the end of the test
+    value = MOI.get(model, MOI.AbsoluteGapTolerance())
+    MOI.set(model, MOI.AbsoluteGapTolerance(), 1e-2)
+    @test MOI.get(model, MOI.AbsoluteGapTolerance()) == 1e-2
+    MOI.set(model, MOI.AbsoluteGapTolerance(), 100)
```

Comment:
This should probably be `100.0`? The `attribute_value_type` is Union{Nothing,Float64}`.

You should also test setting `nothing`.

Reply:
I put an `Int` on purpose, because I thought some users (including myself) would probably pass an integer at some point, and I thought the code should be able to support the conversion.

What would setting the attribute to `nothing` mean? I put `Union{Nothing,Float64}` as attribute type mainly because that's what the other attributes were doing.

Reply:
> What would setting the attribute to nothing mean?

Setting `nothing` would re-set to the default. This should probably be documented in the docstring. If you don't want that, then remove `Nothing` from the attribute value type.

> I put an Int on purpose, because I thought some users (including myself) would probably pass an integer at some point, and I thought the code should be able to support the conversion.

Hmm. This seems problematic. I'd prefer we forced the user to pass `Float64`. If you want to let them pass an `Int`, then the attribute value type should be `Union{Float64,Int}`

Reply:
I can follow the convention of the other attributes, and update the docs to mention that `nothing` uses solver defaults.

However, when I look at the Gurobi wrapper, setting an attribute like, e.g., `TimeLimitSec` to `nothing` is not supported
```julia
using Gurobi
using MathOptInterface
const MOI = MathOptInterface

grb = Gurobi.Optimizer()
MOI.set(grb, MOI.TimeLimitSec(), 1.0)  # no issue
MOI.set(grb, MOI.TimeLimitSec(), nothing)  # not supported
```

Reply:
Re testing `nothing`: this is not done when testing the other attributes, e.g., when testing `TimeLimitSec` or `NumberOfThreads`.
The `setup_test` function does set the attribute to `nothing`, but it's only defined for a mock optimizer (I did the same as for other attributes)

Reply:
Ah. If it's not tested, then that's probably why Gurobi doesn't support it.

Reply:
If I add a test for `nothing`, then downstream solvers will have to support that. It came as a surprise to me, but there's no way that I know of, in Gurobi, to reset a specific parameter to its default value. As far as I can tell, the only two options are to reset all parameters to default, or to set the parameter to a value that happens to be the default. The issue being that default parameter values may change from one release to the next (it's unlikely, but it might).

I don't think it's a good idea to explicitly test for `nothing` values here.

Reply:
Okay. I'm okay not setting `nothing`. We should remove from the attribute value type then.

Reply:
So.... removing `Nothing` as attribute type fails the `_test_attribute_value_type` when testing against `MOIU.MockOptimizer`.
I guess it's because  `MOI.get(mock::MOI.MockOptimizer, attr::MOI.AbstractOptimizerAttribute)` will return `nothing` if said attribute has not been set yet.

This leaves the following options:
* we keep `Union{Nothing,Float64}` as attribute type and...
  * ... we don't test `set/get` with `nothing` input
  * ... we test `set/get` with `nothing` input, and deal with the consequences in downstream solvers
* we change the default behavior of `MockOptimizer{MT}` so that it returns a `Float` when queried for those attributes. I don't know about the consequences of that.

I think the first option (keeping `Nothing` but not testing the `set/get` with `nothing` as input) is the most viable. It's what is currently done for other attributes like `TimeLimitSec` and `NumberOfThreads`.

Reply:
Okay. I guess setting an option to `nothing` isn't a common operation that is requested.

Reply:
The only place I can think of it happening is when passing model around, if the optimizer attributes are copied too. Not even sure that would be the case.

On the other hand, if someone tries to set those attributes to `nothing` in, e.g., CPLEX or Gurobi, they will get an error message because the `MOI.set` functions will require a `::Real` argument (that's the case for, e.g., `TimeLimitSec`). It won't silently fail nor cause segfaults.

The fact that nobody has reported such issue so far indeed suggests that it's not a common operation :)

Reply:
Fair enough. I think the actionable thing I'd like to see is `100` become `100.0`.